File: 1.java
import java.io.*;
import java.util.Scanner;

public class Tree {
    private static int m = 0;
    private static int adj[][];
    private static boolean used[];

    private static void dfs(int v) {
        used[v] = true;
        m++;
        for (int i = 0; i < adj.length; i++) {
            if (adj[v][i] == 1 && !used[i]) {
                dfs(i);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(new FileReader("input.txt"));

        int n = scanner.nextInt();
        adj = new int[n][n];
        used = new boolean[n];

        int edges = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                adj[i][j] = scanner.nextInt();
                edges += adj[i][j];
            }
        }
        scanner.close();

        dfs(0);
        edges /= 2;

        try (BufferedWriter writer = new BufferedWriter(new FileWriter("output.txt"))) {
            if (edges == n - 1 && m == n) {
                writer.write("Yes");
            } else {
                writer.write("No");
            }
        }
    }
}

File: 1.py
import queue
from collections import defaultdict


def dfs(v):
    if used[v]:
        return

    used[v] = True
    global m
    m += 1
    for w in adj[v]:
        dfs(w)


with open("input.txt", "r") as fin:
    n = int(fin.readline())
    edges = 0

    adj = defaultdict(list)
    for i, line in enumerate(fin.readlines()):
        for j, el in enumerate(line.strip().split()):
            if int(el) == 1:
                adj[i].append(j)
                edges += 1

used = [False] * n
m = 0

for vertex in range(n):
    dfs(vertex)
edges //= 2

with open("output.txt", "w") as fout:
    if edges == n - 1 and m == n:
        fout.write("Yes")
    else:
        fout.write("No")
